src/sidebar/util/oauth-client.js

```diff
@@ -245,6 +212,41 @@ class OAuthClient {
     };
     return this.$http.post(url, data, requestConfig);
   }
+
+  /**
+   * Create a pop-up window for use with `OAuthClient#authorize`.
+   *
+   * This function _must_ be called in the same turn of the event loop as the
+   * button or link which initiates login to avoid triggering the popup blocker
+   * in certain browsers. See https://github.com/hypothesis/client/issues/534
+   * and https://github.com/hypothesis/client/issues/535.
+   *
+   * @param {Window} $window - The parent of the popup window.
+   * @return {Window}
+   */
+  static createLoginPopupWindow($window) {
+    // In Chrome & Firefox the sizes passed to `window.open` are used for the
+    // viewport size. In Safari the size is used for the window size including
+    // title bar etc. There is enough vertical space at the bottom to allow for
+    // this.
+    //
+    // See https://bugs.webkit.org/show_bug.cgi?id=143678
+    var width  = 475;
```

Comment:
Completely orthogonal to this discussion, but I'm curious as to where these hard dimensions arose from (in terms of flexibility for different viewports). You can ignore this question for now if you like!

Reply:
These are preferred sizes if the user agent allows, to make the popup look like Dawa's designs, with some allowance for differences in behaviour between Safari and Chrome.

The rendered HTML page is responsive, so it will adapt to other sizes on eg. mobile browsers where only full-screen windows are permitted.